captainkube/azure-pipelines.yml

```diff
@@ -0,0 +1,41 @@
+pool:
+  vmImage: 'Ubuntu 16.04'
+
+
+variables:
+  registryServerName: '$(registryName).azurecr.io'
+  projectName: 'captainkube'
+  imageName: '$(projectName)'
+  # define 4 more variables registryAzureSubscriptionServiceEndpoint, registryName, registryLogin and registryPassword in the build pipeline in UI
+
+
+steps:
+- bash: docker build -f $(system.defaultWorkingDirectory)/$(projectName)/Dockerfile -t $(registryServerName)/$(imageName):$(build.buildId) -t $(registryServerName)/$(imageName):latest $(system.defaultWorkingDirectory)/$(projectName)
+  displayName: 'docker build'
+
+- bash: docker login $(registryServerName) -u $(registryLogin) -p $(registryPassword)
+  displayName: 'docker login'
+
+- bash: |
+    docker push $(registryServerName)/$(imageName):$(build.buildId)
+    docker push $(registryServerName)/$(imageName):latest
+  displayName: 'docker push'
+  condition: and(succeeded(), eq(variables['build.sourceBranch'], 'refs/heads/master'))
+
+- task: HelmInstaller@0
+  displayName: 'Install Helm 2.11.0'
+  inputs:
+    helmVersion: 2.11.0
+
+- bash: helm init --client-only --wait
+  displayName: 'helm init'
+
+- bash: helm package --version $(build.buildId) --destination $(build.artifactStagingDirectory) $(system.defaultWorkingDirectory)/$(projectName)/charts/$(projectName)
```

Comment:
I'm curious how the --version would work here. As far as I know, this expects a `semver` while `build.buildId` doesn't generate a semVer.

Reply:
Good question. It looks like usually it's semver but you could set the version as a non sember too.
```
az acr helm list -n  acr-name -o table
NAME       CHART VERSION    APP VERSION    DESCRIPTION
---------  ---------------  -------------  ------------------------------------------------------------
parrot     564                             Parrot is Captain Kube's sidekick, who shows everything t...
```

So it's easy to change `--version $(build.buildId)` by `--version 1.0.$(build.buildId)`.